File: python_examples/python_examples/pose_goal.py
#!/usr/bin/env python3

# Generic ROS libraries
import rclpy
from rclpy.logging import get_logger

# MoveIt python library
from moveit.planning import (
    MoveItPy,
)

def main():

    # Initialize rclpy and ROS logger
    rclpy.init()
    logger = get_logger("moveit_py.pose_goal")

    # Instantiate MoveItPy and get planning component
    panda = MoveItPy(node_name="moveit_py")
    panda_arm = panda.get_planning_component("panda_arm")

    # set plan start state to current state
    panda_arm.set_start_state_to_current_state()

    # Create PoseStamped message that will hold the target Pose
    from geometry_msgs.msg import PoseStamped
    pose_goal = PoseStamped()
    pose_goal.header.frame_id = "panda_link0"

    # Describe the target pose for the end-effector
    pose_goal.pose.orientation.w = 1.0
    pose_goal.pose.position.x = 0.28
    pose_goal.pose.position.y = -0.2
    pose_goal.pose.position.z = 0.5

    # Set the target pose
    panda_arm.set_goal_state(pose_stamped_msg=pose_goal, pose_link="panda_link8")

    # Create a plan to the target pose
    plan_result = panda_arm.plan()

    # If the plan is successful, get the trajectory and execute the plan
    if plan_result:
        robot_trajectory = plan_result.trajectory
        panda.execute(robot_trajectory, blocking=True, controllers=[])
    else:
        logger.error("Planning failed")

    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: cpp_examples/src/pose_goal.cpp
#include <memory>

#include <rclcpp/rclcpp.hpp>
#include <moveit/move_group_interface/move_group_interface.h>

int main(int argc, char * argv[])
{
  // Initialize ROS and create the Node
  rclcpp::init(argc, argv);
  auto const node = std::make_shared<rclcpp::Node>("pose_goal");

  // Create a ROS logger
  auto const logger = rclcpp::get_logger("pose_goal");

  // Create the MoveIt Move Group Interface for panda arm
  using moveit::planning_interface::MoveGroupInterface;
  auto move_group_interface = MoveGroupInterface(node, "panda_arm");

  // Create a target Pose for the end-effector
  geometry_msgs::msg::Pose target_pose;
  target_pose.orientation.w = 1.0;
  target_pose.position.x = 0.28;
  target_pose.position.y = -0.2;
  target_pose.position.z = 0.5;

  // Set the target pose
  move_group_interface.setPoseTarget(target_pose);

  // Create a plan to that target pose and check if that plan is successful
  moveit::planning_interface::MoveGroupInterface::Plan my_plan;
  bool success = (move_group_interface.plan(my_plan) == moveit::core::MoveItErrorCode::SUCCESS);

  // If the plan is successful, execute the plan
  if(success) {
    move_group_interface.execute(my_plan);
  } else {
    RCLCPP_ERROR(logger, "Planing failed!");
  }

  // Shutdown
  rclcpp::shutdown();
  return 0;
}
